CMakeLists.txt

```diff
@@ -640,9 +653,28 @@ function(JOIN VALUES GLUE OUTPUT)
   set (${OUTPUT} "${_TMP_STR}" PARENT_SCOPE)
 endfunction()
 
-# Set up variables, then write to pkgconfig file
-JOIN("${OSRM_DEFINES}" " " OSRM_DEFINES_STRING)
-JOIN("-I${OSRM_INCLUDE_PATHS}" " -I" OSRM_INCLUDE_PATHS_STRING)
+JOIN("${OSRM_DEFINES}" " " TMP_OSRM_DEFINES)
+set(LibOSRM_CXXFLAGS "${OSRM_CXXFLAGS} ${TMP_OSRM_DEFINES}")
+
+if(BUILD_AS_SUBPROJECT)
+  set(LibOSRM_CXXFLAGS "${LibOSRM_CXXFLAGS}" PARENT_SCOPE)
+  set(LibOSRM_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/include" PARENT_SCOPE)
+  set(LibOSRM_LIBRARY_DIR "${CMAKE_CURRENT_BINARY_DIR}" PARENT_SCOPE)
```

Comment:
This should already be globally available as OSRM_BINARY_DIR (created by the project() call).